### PR TITLE
Refactor nb diff viewmodel code to address leaks

### LINKED ISSUE #224945: [11752] potential listener LEAK detected, having 1055 listeners already. MOST frequent listener (881):
```javascript
Error
at c.create in src/vs/base/common/event.ts:902:15
at p.q [as onDidChangeOutputs] in src/vs/base/common/event.ts:1109:34
at new E in src/vs/workbench/contrib/notebook/browser/diff/diffNestedCellViewModel.ts:94:33
at f.o in src/vs/platform/instantiation/common/instantiationService.ts:162:18
at f.createInstance in src/vs/platform/instantiation/common/instantiationService.ts:128:18
at W.computeModifiedLCS in src/vs/workbench/contrib/notebook/browser/diff/notebookDiffEditor.ts:765:26
at W.computeDiff in src/vs/workbench/contrib/notebook/browser/diff/notebookDiffEditor.ts:696:47
at W.updateLayout in src/vs/workbench/contrib/notebook/browser/diff/notebookDiffEditor.ts:558:67
```
[Go to Errors Site](https://errors.code.visualstudio.com/card?ch=f1e16e1e6214d7c44d078b1f0607b2388f29d729&bH=5af8f6b6-b40a-887f-1650-fe2742aed627)

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #224945

### CODE REVIEW

File: src/vs/workbench/contrib/notebook/browser/diff/diffElementViewModel.ts [hunk 1]
```diff
@@ -219,6 +219,12 @@ export abstract class DiffElementCellViewModelBase extends DiffElementViewModelB
 		this.metadataFoldingState = PropertyFoldingState.Collapsed;
 		this.outputFoldingState = PropertyFoldingState.Collapsed;
 
+		if (original) {
+			this._register(original);
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/browser/diff/diffElementViewModel.ts:223]: leak

File: src/vs/workbench/contrib/notebook/browser/diff/diffElementViewModel.ts [hunk 2]
```diff
@@ -482,8 +488,8 @@ export class SideBySideDiffElementViewModel extends DiffElementCellViewModelBase
 			editorEventDispatcher,
 			initData);
 
-		this.original = original;
-		this.modified = modified;
+		this.original = this._register(original);
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/browser/diff/diffElementViewModel.ts:491]: leak

File: src/vs/workbench/contrib/notebook/browser/diff/notebookDiffEditor.ts [hunk 3]
```diff
@@ -554,205 +558,32 @@ export class NotebookTextDiffEditor extends EditorPane implements INotebookTextD
...
 
 		if (selections) {
 			this._list.setFocus(selections);
 		}
 	}
 
-	/**
-	 * making sure that swapping cells are always translated to `insert+delete`.
-	 */
-	static prettyChanges(model: INotebookDiffEditorModel, diffResult: IDiffResult) {
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/browser/diff/notebookDiffEditor.ts:593]: Moved into separate file

File: src/vs/workbench/contrib/notebook/test/browser/notebookDiff.test.ts [hunk 4]
```diff
@@ -54,19 +86,39 @@ suite('NotebookCommon', () => {
 				modifiedLength: 1
 			}]);
 
-			const eventDispatcher = disposables.add(new NotebookDiffEditorEventDispatcher());
-			const diffViewModels = NotebookTextDiffEditor.computeDiff(accessor, configurationService, model, eventDispatcher, {
-				cellsDiff: diffResult
-			}, undefined);
-			assert.strictEqual(diffViewModels.viewModels.length, 1);
-			assert.strictEqual(diffViewModels.viewModels[0].type, 'modified');
-			diffViewModels.viewModels.forEach(vm => {
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/test/browser/notebookDiff.test.ts:63]: Disposing the new class should now suffice

File: src/vs/workbench/contrib/notebook/test/browser/notebookDiffViewModel.test.ts [hunk 5]
```diff
@@ -1,404 +0,0 @@
-/*---------------------------------------------------------------------------------------------
- *  Copyright (c) Microsoft Corporation. All rights reserved.
- *  Licensed under the MIT License. See License.txt in the project root for license information.
- *--------------------------------------------------------------------------------------------*/
-
-import assert from 'assert';
-import * as sinon from 'sinon';
-import { CellKind } from 'vs/workbench/contrib/notebook/common/notebookCommon';
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/test/browser/notebookDiffViewModel.test.ts:8]: All of these tests are pretty much duplicates of tests in `notebookDiff.test.ts.`
Updated `notebookDiff.test.ts` to include the missing tests

**Review by rebornix — APPROVED** [review]

**Pull request merged**